pkg/model/firewall.go

```diff
@@ -445,3 +470,102 @@ func generateName(o *awstasks.SecurityGroupRule) string {
 	return fmt.Sprintf("from-%s-%s-%s-%dto%d-%s", src, direction,
 		proto, fi.Int64Value(o.FromPort), fi.Int64Value(o.ToPort), dst)
 }
+
+func (b *FirewallModelBuilder) getExistingRulesFromCloud() ([]*awstasks.SecurityGroupRule, error) {
+	tasks := []*awstasks.SecurityGroupRule{}
```

Comment:
Consider moving this line down to just before line 480, then:
```suggestion
	// Estimate the number of rules per security group.
	tasks := make([]*awstasks.SecurityGroupRule{}, 0, len(sgs)*4)
```